src/metabase/api/card.clj

```diff
@@ -273,9 +274,11 @@
   (run-query-for-card card-id parameters))
 
 (defendpoint POST "/:card-id/query/csv"
-  "Run the query associated with a Card, and return its results as CSV."
-  [card-id :as {{:keys [parameters]} :body}]
-  (dataset-api/as-csv (run-query-for-card card-id parameters)))
+  "Run the query associated with a Card, and return its results as CSV. Note that this expects the query as serialized JSON in the 'query' parameters."
+  [card-id query]
+  {query (s/maybe su/JSONString)}
+  (dataset-api/as-csv (run-query-for-card card-id (when query
```

Comment:
You don't actually need to wrap it in the when, parse-string will return nil if it's input is nil